## Introduction to redundancy analysis (RDA)

As part of a search for signatures of selection in my study species (also see fst-outliers.md in this repository), I used a multivariate approach to find associations between environmental variables and the genetic data. RDA has been suggested as a good tool for identifying polygenic selection (Harrisson et al. 2017, Forester et al. 2018). 
A lot of the code I used was based on a workshop presented by Brenna Forester and a lot of the credit for the code goes to her. This is just a guide to fill in some of the gaps regarding preparing data and troubleshooting based on my experience. 

### Imputing missing genotypes the long way round...

RDA requires full datasets so missing values in your genetic matrix need to be imputed. This is where the fun starts.  Since my SNP dataset was generated by DArTseq (Diversity Arrays Technologies Ltd), and I did not call SNPs de novo myself, I only have genotypes (not sequences or read depth). Following the file-conversion.md guide I generated a vcf file. I wanted to use a k-nearest neighbour imputation (LD-kNNi) approach in LinkImpute (Money et al., 2015) but this software is a bit picky. The 


### References
> Forester, B.R., Lasky, J.R., Wagner, H.H., Urban, D.L., 2018. Comparing methods for detecting multilocus adaptation with multivariate genotype–environment associations. Mol. Ecol. 27, 2215–2233. https://doi.org/10.1111/mec.14584

> Harrisson, K.A., Amish, S.J., Pavlova, A., Narum, S.R., Meaghan, M.T., Jarod, L.R., Tonkin, Z., Gilligan, D.M., Ingram, B.A., Sunnucks, P., 2017. Signatures of polygenic adaptation associated with climate across the range of a threatened fish species with high genetic connectivity. Mol. Ecol. 26, 6253–6269. https://doi.org/10.1111/mec.14368

